File: frontend/src/Components/Messages/dto/MessageChatDto.ts
import {isFalsy} from "../../../Validators/IsFalsyObjectValidator";

class MessageChatDto
{
    public messageChatId:string;
    public senderUsername:string;
    public readByOwner:boolean;
    public latestMessage:string;

    public constructor(arg:object)
    {
        if(!isFalsy(arg))
        {
            if(!isFalsy(arg["messageChatId"]))
               this.messageChatId = arg["messageChatId"];
            if(!isFalsy(arg["senderUsername"]))
                this.senderUsername = arg["senderUsername"];
            if(!isFalsy(arg["readByOwner"]))
                this.senderUsername = arg["readByOwner"];
        }
    }
}

export {MessageChatDto};

File: backend/src/main/java/com/svelteup/app/backend/notificationmessage/dtos/message/messagechat/MessageChatDto.java
package com.svelteup.app.backend.notificationmessage.dtos.message.messagechat;

import com.svelteup.app.backend.notificationmessage.models.message.MessageChat;
import io.swagger.annotations.ApiModel;

import java.util.UUID;

@ApiModel(description = "MessageChatDto represents JSON used to transmit MessageChat information to the frontn end.")
public class MessageChatDto
{
    public UUID messageChatId;
    public String senderUsername;
    public Boolean readByOwner;
    public String latestMessage;


    public MessageChatDto(String authenticatedUser, MessageChat messageChat)
    {
        this.messageChatId = messageChat.getSurrogateId();
        if(messageChat.isOwningUser(authenticatedUser))
        {
            this.readByOwner = messageChat.getReadBySecondaryOwningUsername();
            this.senderUsername = messageChat.getSecondaryOwningUsername();
        }
        else
        {
            this.readByOwner = messageChat.getReadByOwningUsername();
            this.senderUsername = messageChat.getOwningUsername();
        }
        this.latestMessage = messageChat.getLatestMessage();
    }
}
